third_party/terraform/tests/resource_dataproc_cluster_test.go.erb

```diff
@@ -54,6 +54,81 @@ func TestExtractInitTimeout_empty(t *testing.T) {
 	t.Fatalf("Expected an error with message '%s', but got %v", expected, err.Error())
 }
 
+func TestDataprocParseImageVersion(t *testing.T) {
+	t.Parallel()
+
+	testCases := map[string]dataprocImageVersion{
+		"1.2":             {"1", "2", "", ""},
+		"1.2.3":           {"1", "2", "3", ""},
+		"1.2.3rc":         {"1", "2", "3rc", ""},
+		"1.2-debian9":     {"1", "2", "", "debian9"},
+		"1.2.3-debian9":   {"1", "2", "3", "debian9"},
+		"1.2.3rc-debian9": {"1", "2", "3rc", "debian9"},
+	}
+
+	for v, expected := range testCases {
+		actual, err := parseDataprocImageVersion(v)
+		if actual.major != expected.major {
+			t.Errorf("parsing version %q returned error: %v", v, err)
+		}
+		if err != nil {
+			t.Errorf("parsing version %q returned error: %v", v, err)
+		}
+		if actual.minor != expected.minor {
+			t.Errorf("parsing version %q returned error: %v", v, err)
+		}
+		if actual.subminor != expected.subminor {
+			t.Errorf("parsing version %q returned error: %v", v, err)
+		}
+		if actual.osName != expected.osName {
+			t.Errorf("parsing version %q returned error: %v", v, err)
+		}
+	}
+
+	errorTestCases := []string{
+		"",
+		"1",
+		"notaversion",
+		"1-debian",
+	}
+	for _, v := range errorTestCases {
+		if _, err := parseDataprocImageVersion(v); err == nil {
+			t.Errorf("expected parsing invalid version %q to return error", v)
+		}
+	}
+}
+
+func TestDataprocDiffSuppress(t *testing.T) {
+	t.Parallel()
+
+	doSuppress := [][]string{
+		{"1.3.10-debian9", "1.3"},
```

Comment:
IIUC a user can specify a less specific version in config and get a more specific version back from the API, in which case we diff suppress. But the inverse wouldn't suppress. If this is the case perhaps a inverse test in noSuppress is worth while.

Reply:
done!